### PR TITLE
Added list collections api

### PR DESCRIPTION
**Pull request by S1LV3RJ1NX**

### CODE REVIEW

File: backend/modules/metadata_store/truefoundry.py [hunk 1]
```diff
@@ -201,6 +201,10 @@ def get_collections(self) -> List[Collection]:
             collection = Collection.parse_obj(self._get_entity_from_run(run=ml_run))
             collections.append(self._polulate_collection(collection))
         logger.debug(f"[Metadata Store] Listed {len(collections)} collections")
+        end = time.time()
+        print(
```
Comment (akashg3627) [inline backend/modules/metadata_store/truefoundry.py:205]: should we remove this now?

File: backend/modules/metadata_store/truefoundry.py [hunk 2]
```diff
@@ -529,3 +533,31 @@ def log_errors_for_data_ingestion_run(
...
+
+        data_sources = []
+        for run in ml_runs:
+            run_params = run.get_params()
+            data_sources.append(
+                {
+                    "type": run_params.get("data_source_fqn").split("::")[0],
+                    "uri": run_params.get("data_source_fqn").split("::")[1],
+                    "fqn": run_params.get("data_source_fqn"),
+                    "metadata": run_params.get("metadata", {}),
```
Comment (akashg3627) [inline backend/modules/metadata_store/truefoundry.py:560]: this metadata field will never be presend, so you can remove that

File: backend/server/routers/data_source.py [hunk 3]
```diff
@@ -9,13 +9,11 @@
 
 
 @router.get("/")
-def list_data_source():
+async def list_data_source():
```
Comment (akashg3627) [inline backend/server/routers/data_source.py:12]: you will need complete details api too

Comment (akashg3627) [reply backend/server/routers/data_source.py:12]: So, better expose this as new api

File: backend/server/routers/collection.py [hunk 4]
```diff
@@ -35,6 +35,16 @@ def get_collections():
         raise HTTPException(status_code=500, detail=str(exp))
 
 
+@router.get("/list-collections")
```
Comment (akashg3627) [inline backend/server/routers/collection.py:38]:
```suggestion
@router.get("/list")
```

File: backend/modules/metadata_store/truefoundry.py (hunk 1 above)
Comment (S1LV3RJ1NX) [reply backend/modules/metadata_store/truefoundry.py:205]: Done.

File: backend/modules/metadata_store/truefoundry.py (hunk 2 above)
Comment (S1LV3RJ1NX) [reply backend/modules/metadata_store/truefoundry.py:560]: done.

File: backend/server/routers/collection.py (hunk 4 above)
Comment (S1LV3RJ1NX) [reply backend/server/routers/collection.py:38]: done.

File: backend/server/routers/data_source.py (hunk 3 above)
Comment (S1LV3RJ1NX) [reply backend/server/routers/data_source.py:12]: Done.

**Review by akashg3627 — APPROVED** [review]

**Pull request merged**